### ISSUE TITLE
Partition TSV files in S3 by year and month as well

### ISSUE THREAD

**Issue by AetherUnbound** [WordPress/openverse-catalog#449]
## Problem

Our TSV files in S3 are currently partitioned by media type, then provider. This is useful for now, but as we get more and more TSVs it will become difficult to find a particular one.

![image](https://user-images.githubusercontent.com/10214785/160666381-87cd31b6-88ee-4fa5-89f8-b8741e432ade.png)

## Description

We should partition the key by year & month as well, that way we only have a maximum of 31 files under a given prefix. In many cases this will lead to a single file nested in a few directories, but that seems a reasonable alternative.

## Additional context

We could partition differently depending on the schedule interval of the provider scripts. For instance, daily & weekly DAGs could be partitioned by `year/month`, while monthly DAGs could just be partitioned by `year`.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

**Comment by sarayourfriend** [comment]
There are apparently some drawbacks to writing S3 keys like this.

https://www.cyclic.sh/posts/aws-s3-why-sometimes-you-should-press-the-100k-dollar-button

https://www.mikulskibartosz.name/partitioning-s3-data-by-date/

I don't believe these apply to us now but wanted to bring it up in case it does in the future. The first link in particular seems frightening :sweat_smile: 

Would using the `dt={YYYY-MM-DD}` format work to solve the problem? It seems like that's the recommended solution to the date partitioning problem that doesn't cause problems down the road :thinking:

**Comment by AetherUnbound** [comment]
That's a good point about the more explicit partitioning! And oof, that article :grimacing: the good news is that our S3 storage is not nearly on that scale. Lifecycle policies would be a good think to set up though. It'd be nice to have the TSVs for say, 6-12 months, but after that I don't think it's necessary. Then again, the catalog bucket pales in comparison to our thumbnail bucket :sweat_smile: 

I think we're still better served by `year/month` and/or `year`. That partitioning helps the human operators more, since we're not doing any sort of Athena based querying or aggregate processing on the files currently anyway.

**Comment by zackkrida** [comment]
> [cyclic.sh/posts/aws-s3-why-sometimes-you-should-press-the-100k-dollar-button](https://www.cyclic.sh/posts/aws-s3-why-sometimes-you-should-press-the-100k-dollar-button)

Great article, but features the least-helpful `tl;dr` I've ever seen:

> TLDR: Mere mortals implement best practices for S3 read and write performance, create a million billion s3 objects with their high performance system, try to clean up the really really expensive mess and find themselves in a dimension as vast as space and as timeless as infinity.
